documentation/src/main/asciidoc/topics/marshalling.adoc

```diff
@@ -1,82 +1,55 @@
-
-[source,java]
-----
-include::code_examples/StorageTypeBinary.java[]
+include::config_examples/protostream_marshaller.xml[]
 ----
 
-==== Equality Considerations
-
-When using lazy deserialization/storing as binary, keys and values are wrapped as link:{javadocroot}/org/infinispan/commons/marshall/[WrappedBytes].
-It is this wrapper class that transparently takes care of serialization and deserialization on demand, and internally may have a reference to the object itself being wrapped, or the serialized, byte array representation of this object.
-
-This has a particular effect on the behavior of equality.
-The equals() method of this class will either compare binary representations (byte arrays) or delegate to the wrapped object instance's equals() method, depending on whether both instances being compared are in serialized or deserialized form at the time of comparison.
-If one of the instances being compared is in one form and the other in another form, then one instance is either serialized or deserialized.
-
-This will affect the way keys stored in the cache will work, when `storeAsBinary` is used, since comparisons happen on the key which will be wrapped by a MarshalledValue. Implementers of equals() methods on their keys need to be aware of the behavior of equality comparison, when a key is wrapped as a MarshalledValue, as detailed above.
-
-==== Store-by-value via defensive copying
-The configuration `storeAsBinary` offers the possibility to enable defensive copying, which allows for store-by-value like behaviour.
-
-{brandname} marshalls objects the moment they're stored, hence changes made to object references are not stored in the cache, not even for local caches. This provides store-by-value like behaviour. Enabling `storeAsBinary` can be achieved:
-
-*  Via XML at the Cache level, either under `<*-cache />` or `<default />` elements:
-
-[source,xml,options="nowrap",subs=attributes+]
-----
-include::config_examples/store_as_binary.xml[]
-----
 
 * Programmatically:
 
 [source,java]
 ----
-include::code_examples/StoreAsBinary.java[]
+include::code_examples/ProtoStreamMarshaller.java[]
 ----
 
-== Advanced Configuration
-Internally, {brandname} uses an implementation of link:{javadocroot}/org/infinispan/commons/marshall/Marshaller.html[this Marshaller interface] in order to marshall/unmarshall Java objects so that they're sent other nodes in the grid, or so that they're stored in a cache store, or even so to transform them into byte arrays for lazy deserialization.
-
-By default, {brandname} uses the link:{javadocroot}/org/infinispan/marshall/core/GlobalMarshaller.html[GlobalMarshaller].
-Optionally, {brandname} users can provide their own marshaller, for example:
-
+=== Java Serialization Marshaller
+Java serialization can also be used to marshall your objects. The only requirements are that your Java objects implement
+the link:https://docs.oracle.com/javase/8/docs/api/java/io/Serializable.html[Serializable] interface, the
```

Comment:
should we update to https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/Serializable.html?

Reply:
We still support java 8, so I think we need to stick with the lowest support version.

Reply:
make sense. I've wondered this looking in other places of the docs.